scripts/populate_annex_scientificname.py

```diff
@@ -0,0 +1,106 @@
+from helpers import get_database_connection, get_config, setup_log_file, execute_sql_from_jinja_string, \
+    insert_or_get_scientificnameid
+from csv import reader
+import time
+import logging
+
+FIELDS_ANNEXSCIENTIFICNAME = ('scientificNameId', 'scientificNameInAnnex', 'isScientificName', 'annexCode', 'remarks')
+
+
+def _load_annex_data_from_file(path):
+    """ Read taxa from file with list of taxa (names) contained in official annexes
+
+    Return a dictionary of names and their corrected versions (which are equal to the names if no correction is needed)
+    """
+    with open(path) as csvfile:
+        annex_data = reader(csvfile)
+        annex_scientificnames = []
+        fields = next(annex_data)
+        print("Columns in " + path + ": " + ", ".join(fields))
+        for row in annex_data:
+            scientific_name_corrected = row[2]
+
+            annex_scientificnames.append({'scientificNameId': None,
+                                         'scientificNameInAnnex': row[1],
+                                         'scientificName': scientific_name_corrected,
+                                         'authorship': row[3],
+                                         'isScientificName': (scientific_name_corrected != ''),
+                                         'annexCode': row[0],
+                                         'remarks': row[5]})
+    return annex_scientificnames
+
+
+def populate_annex_scientificname(conn, config_parser, annex_file):
+    """ Populate the table annexscientificname
+
+    If taxa-limit in configuration file is not a empty string but a number n, then the first n taxa are imported into
+    the table
+    """
+    annex_names = _load_annex_data_from_file(path=annex_file)
+
+    message_n_names_in_annex_file = f"Number of taxa listed in official annexes and ordinances: {len(annex_names)}"
+    print(message_n_names_in_annex_file)
+    logging.info(message_n_names_in_annex_file)
+
+    n_taxa_max = config_parser.get('annex_scientificname', 'taxa-limit')
+    if len(n_taxa_max) > 0:
+        n_taxa_max = int(n_taxa_max)
```

Comment:
I'm a bit confused by this part, is it needed because of some data type issues (int vs str) in the configuration?

Reply:
If `n_taxa_max` is empty in configuration, then import all names from annexes, otherwise take the first `n_taxa_max` names. And as everything from configuration is imported as characters, I make an integer of it.